ca/ca.go

```diff
@@ -148,27 +197,62 @@ func NewCertificateAuthorityImpl(
 		})
 	stats.MustRegister(lintErrorCount)
 
+	certProfileCount := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "certificate_profile",
+			Help: "A counter of certificate profiles used for issuance labelled by profile name, hex representation of the profile hash, certificate purpose (precert/cert), and issuance status (success/failure)",
+		}, []string{"profile", "hash", "purpose", "status"})
+	stats.MustRegister(certProfileCount)
+
 	ca = &certificateAuthorityImpl{
-		sa:             sa,
-		pa:             pa,
-		issuers:        issuers,
-		profile:        certificateProfile,
-		validityPeriod: certExpiry,
-		backdate:       certBackdate,
-		prefix:         serialPrefix,
-		maxNames:       maxNames,
-		keyPolicy:      keyPolicy,
-		log:            logger,
-		signatureCount: signatureCount,
-		signErrorCount: signErrorCount,
-		lintErrorCount: lintErrorCount,
-		clk:            clk,
-		ecdsaAllowList: ecdsaAllowList,
+		sa:               sa,
+		pa:               pa,
+		issuers:          issuers,
+		certProfiles:     certProfiles,
+		validityPeriod:   certExpiry,
+		backdate:         certBackdate,
+		prefix:           serialPrefix,
+		maxNames:         maxNames,
+		keyPolicy:        keyPolicy,
+		log:              logger,
+		signatureCount:   signatureCount,
+		signErrorCount:   signErrorCount,
+		lintErrorCount:   lintErrorCount,
+		certProfileCount: certProfileCount,
+		clk:              clk,
+		ecdsaAllowList:   ecdsaAllowList,
 	}
 
 	return ca, nil
 }
 
+// certificateProfileNameExists checks if the given certificate profile name
+// exists in the CA's list of configured certificate profiles. It returns the
+// hash of the profile or an error. The name is guaranteed to be unique, but the
+// profile values may differ between CA nodes and should not be solely relied
+// upon.
+func (ca *certificateAuthorityImpl) certificateProfileNameExists(name string) ([32]byte, error) {
+	profile, ok := ca.certProfiles.byName[name]
+	if !ok {
+		return [32]byte{}, fmt.Errorf("the CA is not capable of using a profile named %s", name)
+	}
+
+	return profile.GetHash(), nil
+}
+
+// certificateProfileHashExists checks if the given certificate profile hash
+// exists in the CA's list of configured certificate profiles. It returns the
+// profile or an error. The hash over the whole profile is guaranteed to be
+// unique.
+func (ca *certificateAuthorityImpl) certificateProfileHashExists(hash [32]byte) (*issuance.Profile, error) {
+	profile, ok := ca.certProfiles.byHash[hash]
+	if !ok {
+		return nil, fmt.Errorf("the CA is not capable of using a profile with hash %d", hash)
+	}
+
+	return profile, nil
+}
```

Comment:
Similar suggestion to the one above. This method is a getter, renaming it can make that clearer to callers.

```suggestion
// getCertificateProfileByHash retrieves a certificate profile by its hash.
// Returns the profile if it exists in the CA's configured profiles, or an error otherwise.
// The hash is a unique identifier for each profile.
func (ca *certificateAuthorityImpl) getCertificateProfileByHash(hash [32]byte) (*issuance.Profile, error) {
	profile, ok := ca.certProfiles.byHash[hash]
	if !ok {
		return nil, fmt.Errorf("no profile with hash %x configured", hash)
	}

	return profile, nil
}
```